api/lib/routes/scalingHistoryHelper.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+function isMissing(value) {
+  return typeof(value) === "undefined" || value === null;
+}
+
+function isInteger(value) {
+  return Number.isInteger(value);
+}
+exports.validateParameter = function(req) {
+  var appId = req.params.guid;
+  var startTime = req.query["start-time"];
+  var endTime = req.query["end-time"];
+  var order = req.query["order"];
+  var page = req.query["page"];
+  var resultsPerPage = req.query["results-per-page"];
+  if (!appId) {
+    return { valid: false, message: "app guid is required" };
+  }
+
+  if (isMissing(startTime)) {
+    return { valid: false, message: "start-time is required" };
+  }
+  startTime = parseInt(startTime);
+  if (!isInteger(startTime)) {
+    return { valid: false, message: "start-time must be an integer" };
+  }
+
+  if (isMissing(endTime)) {
+    return { valid: false, message: "end-time is required" };
+  }
+  endTime = parseInt(endTime);
+  if (!isInteger(endTime)) {
+    return { valid: false, message: "end-time must be an integer" };
+  }
+
+  if (startTime > endTime) {
+    return { valid: false, message: "start-time must be smaller than end-time" };
+  }
+
+  if (isMissing(page)) {
+    return { valid: false, message: "page is required" };
+  }
+  page = parseInt(page);
+  if (!isInteger(page)) {
+    return { valid: false, message: "page must be an integer" };
+  }
+  if (page < 0) {
+    return { valid: false, message: "page must be greater or equal than 0" };
+  }
+
+  if (isMissing(resultsPerPage)) {
```

Comment:
Cloud Foundry another APIs have 'page' and 'results-per-page' query parameters so it's better to follow it as one of Cloud Foundry components.

For example [/v2/apps API](https://apidocs.cloudfoundry.org/263/apps/list_all_apps.html) returns error when page and results-per-page query parameters are specified with 0.


Call API with results-per-page=0
```
$ cf curl /v2/apps?results-per-page=0
{
   "error_code": "UnknownError",
   "description": "An unknown error occurred.",
   "code": 10001
}
```
Call API with results-per-page=1&page=0
```
 $ cf curl /v2/apps?results-per-page=1\&page=0
{
   "error_code": "UnknownError",
   "description": "An unknown error occurred.",
   "code": 10001
}
```

Call API with results-per-page=1&page=1
```
$ cf curl /v2/apps?results-per-page=1\&page=1
{
   "total_results": 9,
   "total_pages": 9,
   "prev_url": null,
   "next_url": "/v2/apps?order-direction=asc&page=2&results-per-page=1",
   "resources": [
      {
         "metadata": {
            "guid": "88ead269-e996-4811-96e6-7f6d6111b026",
            "url": "/v2/apps/88ead269-e996-4811-96e6-7f6d6111b026",
            "created_at": "2017-05-31T01:51:56Z",
            "updated_at": "2017-06-06T03:54:07Z"
         },
   .....
```